tap_pipedrive/stream.py

```diff
@@ -158,11 +159,14 @@ def get_deal_ids(self, tap):
             self.more_ids_to_get = self.more_items_in_collection  # note if there are more pages of ids to get
             self.next_start = self.start  # note pagination for next loop
             ndeals = len(response.json()['data'])
+
+            # check add_time and stage_change_time to catch all deals that have been added or have changed stages
             this_page_ids = [response.json()['data'][i]['id'] for i in range(ndeals) 
-                             if response.json()['data'][i]['stage_change_time'] is not None 
-                             and pendulum.parse(response.json()['data'][i]['stage_change_time']) < self.stream_start
-                             and pendulum.parse(response.json()['data'][i]['stage_change_time']) >= checkpoint]
-            
+                                    if (self.stream_start > pendulum.parse(response.json()['data'][i]['add_time']) >= checkpoint) 
+                                    or (response.json()['data'][i]['stage_change_time'] is not None and 
+                                           self.stream_start > pendulum.parse(response.json()['data'][i]['stage_change_time']) >= checkpoint)]
```

Comment:
At first glance, this is a very dense and complex looking list comprehension. Can we make this more readable and less complex?

I think a helper function would make this section less complex and if we throw in some comments, we could clarify the logic too.

Reply:
Yeah, that list comprehension was pretty dense. I stuck it into a helper function, as requested, and I split it into two pieces with comments on each.